### ISSUE TITLE
Plugin is adding other non-listed photos to gallery

### ISSUE THREAD

**Issue by casspop** [pelican-plugins/photos#68]
It was working great, but my last gallery, which has three photos in its folder, shows fifteen photos when you click on it on the site.

I am happy to provide whatever you need to see in order to try to diagnose this. 
Pelican version 4.8.0
pelican-photos 1.3.0

Active page with gallery problem https://www.drgerg.com/mpd-is-just-awesome.html

Thanks!

[mpd-is-just-awesome.md](https://github.com/pelican-plugins/photos/files/10047569/mpd-is-just-awesome.md)

**Comment by casspop** [comment]
I'm not sure what I did, but now it's working.  It started working again when I added a ```code block``` right before the gallery.

I went back and took the codeblock out, and it's still good.  Weird, but satisfactory.

It may be related to the fact that I did not have:
PHOTO_INLINE_GALLERY_PATTERN = r"gallery::(?P<gallery_name>[/{}\w_-]+)"
actually specified in pelicanconf.py .  I didn't have it in there because it was labeled as 'default', so I didn't think it necessary.

I guess you can close this, even though we really don't know what caused it.

Thanks!

**Comment by casspop** [comment]
Well, it seems I closed that ticket prematurely.  I'm not totally stupid, but I'm baffled.

It worked one time, then the same weird behavior returned.  I tried stepping back through changes, but I never found the magic moment again.

I am aware I need to flush browser buffers between attempts, so I've been doing that regularly.

- I looked for commonalities among the photos the plugin was including in the gallery.  I noticed they were all .png, so I changed them to .jpg.  No diff.
- I noticed they all had underscores and dashes in the filenames, so I cleared all those out. No diff.
- I copied the gallery folder in Pictures and renamed the copy.  Changed name in the .md file.  No diff.
- I played with the PATTERN entry in pelicanconf.py, but saw no difference.
- I upgraded to pelican-photos 1.4.0.  pip list confirms that.

I have several galleries in my blog all working fine.  I can't figure out why this one is not working right.  Any help would be appreciated.

**Comment by casspop** [comment]
Crap!  Now it's working.  Which is great, but damn.  I don't understand why moving it down the page results in working.  I may be caught in a cache/buffer loop that I'm not aware of.  That's the only thing I can think of. 

I just wish I knew what caused it in the first place. 

Any thoughts?

**Comment by casspop** [comment]
OK.  I just noticed a new pattern.  Maybe this is part of the answer.

The affected post is the front page at https://www.drgerg.com .  When you view the gallery from that URL, it shows 15 photos.

If you view the post from this URL: https://www.drgerg.com/mpd-is-just-awesome.html the gallery displays 3 photos, which is correct.

Thoughts?

**Comment by phibos** [comment]
I'm sorry, but I can't find a gallery on the front page and in the article everythings looks ok to me.

**Comment by casspop** [comment]
Hello!

I was afraid that would happen.  The post that was showing that problem has rolled off the front page.

If you would like, I can create a dummy post to replicate the problem.

Let me know when you'd like to look at it.

Thanks!

**Comment by phibos** [comment]
Can you please retest with the latest main branch. An issue with the exclude list has been fixed with PR #75 

But I can't find any issues with your page. I had a look at https://www.drgerg.com/tag/ubuntu.html which should be the same as the front page and should show the 15 images from other galleries you mentioned. But there are just 3 images rendered at the moment. Also on the page https://www.drgerg.com/tag/mpd.html which is the same just for the mpd tag.

**Comment by casspop** [comment]
I will work up a test post this evening.  

Thanks!

**Comment by casspop** [comment]
Monday, Dec. 19, 4:45pm CDT:  I created a test post.  It's the default post on the site.  The behavior is repeated there.

**Comment by phibos** [comment]
Thanks for adding the test post. Now I can see the issue. But it is not an issue with the plugin. It is the way how you designed your template and how the code for the lightbox identifies the images to use.

There are many div-elements with ```class="gallery"``` on the front page. As you can see on the screenshot below these elements are added to posts where only the summary should be visible.

![screenshot](https://user-images.githubusercontent.com/1975606/208751310-45e8fa97-2eac-489e-a46c-eba5cb140b6d.png)

The JavaScript code you use all images from all div-elements with the ```class="gallery"``` are added to the lightbox.

```javascript
$('.gallery').magnificPopup({
	delegate: 'a',
	type: 'image',
```
So the easiest way to remove the images from the lightbox is to clean up the template code and only add the galleries where they should be visible.

**Comment by casspop** [comment]
Thanks for the insights.  I pulled the template code from the photos-plugins docs.  

I will try to wrap my head around what is going on with it.  It still feels odd that it only happens when the post is opened in the main URL, but not when opened with a specific URL.

I appreciate the fact that you looked at it.

**Comment by casspop** [comment]
I finally got time to dig into this, and I figured out what was happening.  I had placed the inline gallery entry too close to the top of the article content.  For example, the 'gallery::{photo}PiNet/shoppi_gallery' entry was at the beginning of the article, so it was getting caught up in the 'article.summary' 

The default length of the 'article.summary' string is set at 50, so all text in the content source file inside that parameter is going to be sent to the templates.  Because index.html contains summaries of all previous articles below the new main article, and some of those summaries contained gallery entries, the galleries were being processed as a unit.

Another way to avoid the problem is to use a 'Summary: Put the summary here.' metadata line in the header of the content file.  That's not something I had been doing, but instead was just letting the automations handle created the summaries for me.

I appreciate @phibos taking the time to give me the hint I needed to move toward the solution.
